pkgs/tools/graphics/netpbm/default.nix

```diff
@@ -27,16 +28,43 @@ stdenv.mkDerivation {
     sha256 = "1m7ks6k53gsjsdazgf22g16dfgj3pqvqy9mhxzlwszv5808sj5w5";
   };
 
+  makeFlags = [
+    "CC=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc"
+  ];
+
   postPatch = ''
     # Install libnetpbm.so symlink to correct destination
     substituteInPlace lib/Makefile \
       --replace '/sharedlink' '/lib'
+  '' + (stdenv.lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+      substituteInPlace buildtools/configure.pl \
+          --replace 'pkg-config' '${buildPackages.pkgconfig}/bin/${buildPackages.pkgconfig.targetPrefix}pkg-config'
+
+      substituteInPlace GNUmakefile \
+          --replace 'pkg-config' '${buildPackages.pkgconfig}/bin/${buildPackages.pkgconfig.targetPrefix}pkg-config'
+
+      substituteInPlace config.mk.in \
+          --replace 'AR = ar' 'AR = ${stdenv.lib.getBin stdenv.cc.bintools.bintools}/bin/${stdenv.cc.targetPrefix}ar'
+      substituteInPlace config.mk.in \
+          --replace 'RANLIB = ranlib' 'RANLIB = ${stdenv.lib.getBin stdenv.cc.bintools.bintools}/bin/${stdenv.cc.targetPrefix}ranlib'
+
+      substituteInPlace buildtools/install.sh \
+          --replace 'STRIPPROG-strip' 'STRIPPROG:-${stdenv.lib.getBin stdenv.cc.bintools.bintools}/bin/${stdenv.cc.targetPrefix}strip'
+    '');
+
+  preBuild = ''
+    pushd buildtools
+    make CC=${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc
```

Comment:
Could you clarify why is this needed? Would not the default `CC = cc` in `config.mk` already be  the same?

```
nix-repl> pkgsCross.aarch64-multiplatform.buildPackages.stdenv.cc.targetPrefix       
""
```

Reply:
Oh, looks like during cross-compilation there are only the prefixed binaries.

Though using `CC` environment variable might work.

Reply:
During cross-platform build, the buildtools subpackage needs to be built with the local native compiler as those tools will be run during build, while the rest of the package will need to be built using the target platform compiler.
That is why I am building them separately like this.

Reply:
Right, I did not notice the directory change.

Actually, it looks like the build tools makefile uses `CC_FOR_BUILD` variable so setting it in `config.mk` should remove the necessity to invoke `make` manually.

Reply:
You are right, it should work, but the build fails for buildtools with correct `CC_FOR_BUILD` and `LD_FOR_BUILD` set in `config.mk`. The workaround in preBuild works, though I haven't investigated any further why `config.mk` doesn't.